src/providers/cryptoauthlib/key_management.rs

```diff
@@ -18,6 +19,13 @@ impl Provider {
         application_identity: &ApplicationIdentity,
         op: psa_generate_key::Operation,
     ) -> Result<psa_generate_key::Result> {
+        if let Err(ResponseStatus::DeprecatedPrimitive) = op.check_deprecated() {
+            warn!("The key requested to generate is deprecated");
+            if !crate::utils::GlobalConfig::allow_deprecated() {
+                return Err(ResponseStatus::DeprecatedPrimitive);
+            }
+        }
```

Comment:
Given that this bit of code is literally the same across all providers, would it be possible to do something a la [`format_error`](https://github.com/parallaxsecond/parsec/blob/main/src/lib.rs#L40), and only call that macro everywhere?

Reply:
Yea, good idea. Where do you think placing this macro would be the best? `src/providers/mod.rs`?
Also, do you think `warn_on_deprecated` or `return_on_deprecated` is okay for the macro names?
I am open to suggestions 😸

Reply:
Yeah, that sounds like a suitable location. In terms of name those sound good as well, or `handle_deprecated` or anything like that, doesn't matter that much since it's "internal"